### PR TITLE
fix: all setpoints in SI (#30)

### PR DESCRIPTION
**Pull request by SebastianDD**
#30

### CODE REVIEW

File: powerfactory_utils/exporter/exporter.py [hunk 1]
```diff
@@ -1923,24 +1922,28 @@ def create_external_grids_ssc_states(
...
+            if g.bus1 is None:
+                logger.warning(f"External grid {name} not connected to any bus. Skipping.")
+                continue
+
             g_type = GridType(g.bustp)
             if g_type == GridType.SL:
                 grid = ExternalGridSSC(
                     name=name,
-                    u_0=g.usetp,
+                    u_0=round(g.usetp * g.bus1.cterm.uknom, DecimalDigits.VOLTAGE) * Exponents.VOLTAGE,
```
Comment (sasanjac) [inline powerfactory_utils/exporter/exporter.py:1933]:
```suggestion
                    u_0=round(g.usetp * g.bus1.cterm.uknom * Exponents.VOLTAGE, DecimalDigits.VOLTAGE),
```

File: powerfactory_utils/exporter/exporter.py [hunk 2]
```diff
@@ -1923,24 +1922,28 @@ def create_external_grids_ssc_states(
...
                     phi_0=g.phiini,
                 )
             elif g_type == GridType.PV:
                 grid = ExternalGridSSC(
                     name=name,
-                    u_0=g.usetp,
-                    p_0=g.pgini,
+                    u_0=round(g.usetp * g.bus1.cterm.uknom, DecimalDigits.VOLTAGE) * Exponents.VOLTAGE,
```
Comment (sasanjac) [inline powerfactory_utils/exporter/exporter.py:1939]:
```suggestion
                    u_0=round(g.usetp * g.bus1.cterm.uknom * Exponents.VOLTAGE, DecimalDigits.VOLTAGE),
```

File: powerfactory_utils/exporter/exporter.py [hunk 3]
```diff
@@ -1309,15 +1309,14 @@ def create_q_controller(
...
+        q_r = s_r * math.sin(math.acos(cosphi_r))
 
         cosphi_type = None
         cosphi = None
         q_set = None
         m_tab2015 = None  # Q(U) droop related to VDE-AR-N 4120:2015
         m_tar2018 = None  # Q(U) droop related to VDE-AR-N 4120:2018
-        qmax_ue = math.tan(math.acos(cosphi_r))  # q_r / p_r  default
+        qmax_ue = q_r
         qmax_oe = qmax_ue
```
Comment (sasanjac) [inline powerfactory_utils/exporter/exporter.py:1320]:
```suggestion
        qmax_oe = q_r
```

File: powerfactory_utils/exporter/exporter.py [hunk 4]
```diff
@@ -1374,8 +1373,8 @@ def create_q_controller(
                 elif controller_type == ControllerType.Q_U:
                     u_nom = round(ext_ctrl.refbar.uknom, DecimalDigits.VOLTAGE) * Exponents.VOLTAGE  # voltage in V
 
-                    qmax_ue = abs(ext_ctrl.Qmin / p_r)  # per unit
-                    qmax_oe = abs(ext_ctrl.Qmax / p_r)  # per unit
+                    qmax_ue = abs(ext_ctrl.Qmin)
```
Comment (sasanjac) [inline powerfactory_utils/exporter/exporter.py:1376]:
```suggestion
                    qmax_ue = round(abs(ext_ctrl.Qmin), DecimalDigits.POWER)
```

File: powerfactory_utils/exporter/exporter.py [hunk 5]
```diff
@@ -1374,8 +1373,8 @@ def create_q_controller(
...
+                    qmax_oe = abs(ext_ctrl.Qmax)
```
Comment (sasanjac) [inline powerfactory_utils/exporter/exporter.py:1377]:
```suggestion
                    qmax_oe = round(abs(ext_ctrl.Qmax), DecimalDigits.POWER)
```

File: powerfactory_utils/exporter/exporter.py [hunk 6]
```diff
@@ -1332,8 +1331,8 @@ def create_q_controller(
             elif controller_type == ControllerType.Q_CONST:
                 q_set = gen.qgini
             elif controller_type == ControllerType.Q_U:
-                qmax_ue = abs(gen.Qfu_min / p_r)  # p.u.
-                qmax_oe = abs(gen.Qfu_max / p_r)  # p.u.
+                qmax_ue = abs(gen.Qfu_min)
```
Comment (sasanjac) [inline powerfactory_utils/exporter/exporter.py:1334]:
```suggestion
                qmax_ue = round(abs(gen.Qfu_min), DecimalDigits.POWER)
```

File: powerfactory_utils/exporter/exporter.py [hunk 7]
```diff
@@ -1332,8 +1331,8 @@ def create_q_controller(
...
+                qmax_oe = abs(gen.Qfu_max)
```
Comment (sasanjac) [inline powerfactory_utils/exporter/exporter.py:1335]:
```suggestion
                qmax_oe = round(abs(gen.Qfu_max), DecimalDigits.POWER)
```

**Review by sasanjac — CHANGES_REQUESTED** [review]

File: powerfactory_utils/exporter/exporter.py [hunk 8]
```diff
@@ -1309,15 +1309,14 @@ def create_q_controller(
         }
         s_r = gen.sgn
         cosphi_r = gen.cosn
-        p_r = s_r * cosphi_r
-        # q_r = s_r * math.sin(math.acos(cosphi_r))
+        q_r = s_r * math.sin(math.acos(cosphi_r))
```
[the remark this answers is not in the document]
Comment (SebastianDD) [reply powerfactory_utils/exporter/exporter.py:1312]: decline. rounding is intended at the end of the function, right?
`q_r `is oinly used for `qmax` and `q_max` is rounded at the end.

File: powerfactory_utils/exporter/exporter.py (hunk 6 above)
Comment (SebastianDD) [reply powerfactory_utils/exporter/exporter.py:1334]: decline. rounding is intended at the end of the function, right?
q_r is oinly used for qmax and q_max is rounded at the end.

File: powerfactory_utils/exporter/exporter.py (hunk 7 above)
Comment (SebastianDD) [reply powerfactory_utils/exporter/exporter.py:1335]: decline. rounding is intended at the end of the function, right?
q_r is oinly used for qmax and q_max is rounded at the end.

File: powerfactory_utils/exporter/exporter.py (hunk 4 above)
Comment (SebastianDD) [reply powerfactory_utils/exporter/exporter.py:1376]: decline. rounding is intended at the end of the function, right?
q_r is oinly used for qmax and q_max is rounded at the end.

File: powerfactory_utils/exporter/exporter.py (hunk 5 above)
Comment (SebastianDD) [reply powerfactory_utils/exporter/exporter.py:1377]: decline. rounding is intended at the end of the function, right?
q_r is oinly used for qmax and q_max is rounded at the end.

**Review by sasanjac — APPROVED** [review]

**Pull request merged**